### PR TITLE
Learnmsa align

### LINKED ISSUE #3814: new module: learnMSA/align
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by luisas**
Closes #3814 

## PR checklist
- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/modules/nf-core/learnmsa/align/main.nf [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+//               where "task.ext" is a string.
+//               Any parameters that need to be evaluated in the context of a particular sample
+//               e.g. single-end/paired-end data MUST also be defined and evaluated appropriately.
+// TODO nf-core: Software that can be piped together SHOULD be added to separate module files
+//               unless there is a run-time, storage advantage in implementing in this way
+//               e.g. it's ok to have a single module for bwa to output BAM instead of SAM:
+//                 bwa mem | samtools view -B -T ref.fasta
+// TODO nf-core: Optional inputs are not currently supported by Nextflow. However, using an empty
+//               list (`[]`) instead of a file can be used to work around this issue.
+
```
Comment (FriederikeHanssen) [inline modules/modules/nf-core/learnmsa/align/main.nf:17]:
```suggestion
```

File: modules/modules/nf-core/learnmsa/align/main.nf [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+process LEARNMSA_ALIGN {
+    tag "$meta.id"
+    label 'process_medium'
+
+    // TODO nf-core: List required Conda package(s).
+    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
+    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
+    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
```
Comment (FriederikeHanssen) [inline modules/modules/nf-core/learnmsa/align/main.nf:25]:
```suggestion
```

File: modules/modules/nf-core/learnmsa/align/main.nf (hunk 1 above)
Comment (maxulysse) [inline modules/modules/nf-core/learnmsa/align/main.nf:17]:
```suggestion
```

**Review by FriederikeHanssen — COMMENTED** [review]
I think you accidentally generated the file twice and have a template one in a new subfolder?

File: modules/nf-core/learnmsa/align/main.nf [hunk 3]
```diff
@@ -0,0 +1,47 @@
+
```
Comment (FriederikeHanssen) [inline modules/nf-core/learnmsa/align/main.nf]:
```suggestion
```

Comment (maxulysse) [inline modules/nf-core/learnmsa/align/main.nf:1]:
```suggestion
```

File: modules/nf-core/learnmsa/align/meta.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  - meta2:
+      type: map
+      description: |
+        Groovy Map containing tree information
+        e.g. `[ id:'test_tree']`
+
+  - tree:
+      type: file
+      description: Input guide tree in Newick format
+      pattern: "*.{dnd}"
```
Comment (maxulysse) [inline modules/nf-core/learnmsa/align/meta.yml:37]: I don't see that as an input in the main.nf

File: modules/nf-core/learnmsa/align/main.nf [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    learnMSA \\
+        $args \\
+        -i $fasta \\
+        -o ${prefix}.aln
```
Comment (FriederikeHanssen) [inline modules/nf-core/learnmsa/align/main.nf:27]: just checking: this tool doesn't take an avaialble cpu number? Couldn't find docs listing all params

Comment (FriederikeHanssen) [reply modules/nf-core/learnmsa/align/main.nf:27]: In that case and if it is single threaded, I would change the ressource label to single

Comment (luisas) [reply modules/nf-core/learnmsa/align/main.nf:27]: Not as far as I know - it should ideally run on GPUs and, per default, will use all the available ones. Apparently, one can specify the name of the GPU devices, but not as a number, just as IDs - and this should probably go in $args.

File: modules/nf-core/learnmsa/align/meta.yml (hunk 4 above)
Comment (luisas) [reply modules/nf-core/learnmsa/align/meta.yml]: You are right! Thanks for spotting this :)

**Review by maxulysse — APPROVED** [review]

**Pull request merged**